tests/static-analysis/fromFile.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+/**
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+include __DIR__ . '/../../vendor/autoload.php';
+
+use loophp\collection\Collection;
+
+/**
+ * @param Collection<int, string> $collection
+ */
+function check(Collection $collection): void
+{
+}
+
+check(Collection::fromFile('https://loripsum.net/api')->limit(25));
```

Comment:
let me know if this makes sense, all I wanted to ensure is that the return type of `fromFile` is properly registered and checked by Psalm & PHPStan as `Collection<int, string>`

Reply:
This seems to be the proper way indeed.